### ISSUE TITLE
Develop the Reward Model Architecture

### ISSUE THREAD

**Issue by RishabhMalviya** [RishabhMalviya/raag-midi-gen#8]
# Completion Criteria
Have a functioning `nn.Module` that takes 2 melody tensors (the `reference_melody` and an `embellished_melody`), and outputs a score for the quality of the `embellished_melody` with respect to the `reference_melody`.

# Ideas
Probably train two encoder models (with the final aggregation being done by taking the mean over all the output contextualized vectors). 

One encoder will encode the reference melody, the other will encode the embellished melody.

Then take a dot product of the two embellishments to get a score. This will have the advantage of keeping the reward between -1 and 1, which should make it much easier to apply a baseline of 0 during the RL training of the melody generator.
